### ISSUE TITLE
Visual Studio compiler not found & files missing (python 2.7.16, scons 3.0.5, VS 2019)

### ISSUE THREAD

**Issue by Kejoko** [godotengine/godot#28152]
**Godot version:**
Master Branch 6697fd9

**OS/device including version:**
Windows 10 64-bit
Python version 2.7.16
Scons version 3.0.5
VS 2019

**Issue description:**
I attempted to build godot using the _x64 Native Tools Command propmt for VS 2019_ When I entered the command "scons platform=windows vsproj=yes" it gave me the following errors:

`D:\!Dev\!Godot\godot>scons platform=windows vsproj=yes`
`scons: Reading SConscript files ...`

`scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly`
`File "D:\!Dev\!Godot\godot\SConstruct", line 3, in <module>`

`scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly`
`File "D:\!Dev\!Godot\godot\SConstruct", line 64, in <module>`
`Configuring for Windows: target=debug, bits=default`
`Using Mingw`
`WindowsError: [Error 2] The system cannot find the file specified:`
`  File "D:\!Dev\!Godot\godot\SConstruct", line 345:`
`          version = methods.get_compiler_version(env)`
`  File "D:\!Dev\!Godot\godot\methods.py", line 663:`
`         version = decode_utf8(subprocess.check_output([env['CXX'], '--version']).strip())`
`  File "C:\Python27\lib\subprocess.py", line 216:`
`         process = Popen(stdout=PIPE, *popenargs, **kwargs)`
`  File "C:\Python27\lib\subprocess.py", line 394:`
`         errread, errwrite)`
`  File "C:\Python27\lib\subprocess.py", line 644:`
`         startupinfo)`

`D:\!Dev\!Godot\godot>`

**Comment by akien-mga** [comment]
Did you install the C++ compiler? https://docs.godotengine.org/en/latest/development/compiling/compiling_for_windows.html#installing-visual-studio-caveats

**Comment by Kejoko** [comment]
Yes, I am fairly positive I did. However, is there a way to check to be certain?

**Comment by Simran-B** [comment]
VS2019 is version 16.0, so there should be a folder
`C:\Program Files (x86)\Microsoft Visual Studio 16.0\VC\`
with a `cl.exe` in it if you installed VS with C++.

BTW. is it intended that scons starts a compilation if `vsproj=yes` instead of just generating the .sln file?

**Comment by aaronfranke** [comment]
Does it work if you use Python 3?

**Comment by Calinou** [comment]
@Kejoko Can you still reproduce this in Godot 3.2 with SCons 3.1.2?

**Comment by swarnimarun** [comment]
Btw I just had a similar issue, I wasted about 4 hours on and off, reseting all my configurations(I use a fairly extensive ps1 configuration file, so I just assumed I messed something up in the past week I hadn't built godot from powershell) and shuffling around and trying to build using different compilers including mingw(which worked), was about to try wsl when figured out the issue.

I realized my version of Scons was old like 3.0.1, and I had upgraded to VS2019 which didn't sit well with it, (basically there wasn't a env setting to pull it), so I just did a pip upgrade of scons and it worked for both versions of python.
I also had VS2015 installed which should have been picked up but it was an internal bug of scons.

Currently I am on 3.1.x version of scons, and python3.7 with Windows 10(insider) and VS2019 (latest).

Anyways,
So yeah we can close it, as it's a scons 3.0.x bug.
Reference to bug,
https://github.com/SCons/scons/issues/3346
Which seems to be fixed in scons 3.1.x.

cc: @Calinou
